src/main/java/com/amazon/opendistroforelasticsearch/sql/rewriter/nestedfield/Identifier.java

```diff
@@ -52,15 +52,15 @@ void rewrite(Scope scope) {
     }
 
     String path() {
-        return separatorIndex() == -1 ? "" : expr.getName().substring(0, separatorIndex());
+        return lastSeparatorIndex() == -1 ? "" : expr.getName().substring(0, lastSeparatorIndex());
     }
 
     String name() {
-        return expr.getName().substring(separatorIndex() + 1);
+        return expr.getName().substring(lastSeparatorIndex() + 1);
```

Comment:
Is it possible to have multi-field in a deeply nested object, e.g. have something like `table1.subobject1.field1.keyword`, where the field name is actually `field1.keyword`, not just `keyword`? I am not saying you should support it with this change, but if yes, please add a TODO comment, or better create an issue of type `Enhancement`.